filters/src/passthrough.cpp

```diff
@@ -273,6 +273,106 @@ pcl::PassThrough<pcl::PCLPointCloud2>::applyFilter (PCLPointCloud2 &output)
   removed_indices_->resize (nr_removed_p);
 }
 
+void
+pcl::PassThrough<pcl::PCLPointCloud2>::applyFilter (std::vector<int> &indices)
+{
+  // The arrays to be used
+  indices.resize (indices_->size ());
+  removed_indices_->resize (indices_->size ());
+  int oii = 0, rii = 0;  // oii = output indices iterator, rii = removed indices iterator
+  const std::uint32_t xyz_offset[3] = {input_->fields[x_idx_].offset, input_->fields[y_idx_].offset, input_->fields[z_idx_].offset};
+  PCL_DEBUG ("[pcl::%s<pcl::PCLPointCloud2>::applyFilter] Field offsets: x: %u, y: %u, z: %u.\n", filter_name_.c_str (), xyz_offset[0], xyz_offset[1], xyz_offset[2]);
+
+  // Has a field name been specified?
+  if (filter_field_name_.empty ())
+  {
+    // Only filter for non-finite entries then
+    for (const auto ii : indices)  // ii = input index
+    {
+      float pt[3];
+      memcpy (&pt[0], &input_->data[ii * input_->point_step + xyz_offset[0]], sizeof(float));
+      memcpy (&pt[1], &input_->data[ii * input_->point_step + xyz_offset[1]], sizeof(float));
+      memcpy (&pt[2], &input_->data[ii * input_->point_step + xyz_offset[2]], sizeof(float));
```

Comment:
:laughing: But what would be a good solution? `data` is a vector of `uint8_t`, so is there another way instead of using memcpy? And that would mean an if-statement here and further down to differentiate between float and double.

Reply:
> `data` is a vector of `uint8_t`, so is there another way instead of using memcpy?

Casting to the right pointer type based on the reported type, the vector holding the fields' list, usually also stores type being passed as an enum type, and then derefence the pointer. It's always accessing a single element. 

https://github.com/PointCloudLibrary/pcl/blob/3e04ea7f5e8dae81db2411f74ba18dc7da448c3d/common/include/pcl/PCLPointField.h#L23-L30
https://github.com/PointCloudLibrary/pcl/blob/3e04ea7f5e8dae81db2411f74ba18dc7da448c3d/common/include/pcl/point_traits.h#L87-L95

> And that would mean an if-statement here and further down to differentiate between float and double.

Well, there's not much we can do here right? In this `PCLPointCloud2` type, all fields all inferred at runtime. So it is only natural that specializations of this type suffer from the runtime penalties associated with it.